ui/layouts/OwnerApp.tsx

```diff
@@ -17,7 +19,11 @@ interface Props {
   title: React.ReactNode
 }
 
-export const OwnerApp: React.FC<Props> = ({ children, title }) => {
+const OwnerAppThing: React.FC<WithOwnerProps & Props> = ({
+  owner,
+  children,
+  title,
+}) => {
   const { data: companies } = useCompanies()
 
```

Comment:
The `withOwner` HOC is only meant for whole pages (component files inside the `pages` directory). But you can directly use the `useOwner` hook in here, instead of using the HOC:

```tsx
export const OwnerApp: React.FC<Props> = ({ children, title }) => {
  const { data: owner } = useOwner()
  const { data: companies } = useCompanies()

  return (
    <Limit>
      // ...
```

Since all pages are already wrapped into the `withOwner` HOC, and the HOC prefetches the owner data before rendering the pages, `owner` will be available immediately without having to wait for it to load.

Reply:
Ahhh nice, I knew there should be a better way! Thanks :)